### PR TITLE
Feat/change dock undock subs to actions

### PR DESCRIPTION
**Pull request by michelleat**
In this PR, I changed the mechanism used to trigger docking or undocking.  Previously, the autodocking node subscribed to /auto_dock/undock and /auto_dock/dock.  Now, the node runs two action servers (dock and undock).  

Once a goal is sent, both actions will send feedback until the docking or undocking is completed.  The feedback currently is set to just be the docking state (a string), so for example if you send an /autodock/dock/goal, you can get feedback such as "searching", "centering", "approaching", until the dock has completed.  The dock action is also able to be cancelled thru the action server.  

Please take a look at execute_dock_cb and execute_undock_cb specifically.   They seem to work fine in preliminary testing, but I wonder if there are any threading concerns.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -38,10 +39,23 @@ add_service_files(
...
+
+generate_messages(
+  DEPENDENCIES 
+  actionlib_msgs
+  std_msgs
+  visualization_msgs
+)
+
 catkin_package(
   LIBRARIES image_sub_nodelet
```
Comment (khrisnakamarga) [inline CMakeLists.txt:57]: is nodelet required by the original docking code?

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# These are some examples of commonly ignored file patterns.
```
Comment (pvishal) [inline .gitignore]: I like having a standard .gitignore that we reuse across projects.

File: CMakeLists.txt (hunk 1 above)
Comment (pvishal) [reply CMakeLists.txt:57]: I have the same question. Unless I am missing something, I don't see this code being loaded as a nodelet.

File: action/dock.action [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#goal
+bool dock
+---
+#result
+bool docked
+---
+#feedback
+string docking_state
```
Comment (pvishal) [inline action/dock.action]: I let the docking state be a string in the code. But for the message, I think it should be an enum. 

We can backport the state machine code to use enum states later but the new interfaces we add should be proper. What are your thoughts @dean4ta @khrisnakamarga ?

File: package.xml [hunk 4]
```diff
@@ -34,6 +34,10 @@
   <build_export_depend>message_runtime</build_export_depend>
   <exec_depend>message_runtime</exec_depend>
   <exec_depend>wibotic_msg</exec_depend>
+  <build_depend>actionlib</build_depend>
+  <build_depend>actionlib_msgs</build_depend>
+  <exec_depend>actionlib</exec_depend>
+  <exec_depend>actionlib_msgs</exec_depend>
```
Comment (pvishal) [inline package.xml]: If a dependency is all of build, exec prefer using the `<depend>` tag.

**Review by pvishal — COMMENTED** [review]
How did you test this action-based interface? Do you have a test node that you called this with or did you just call it on the terminal?

**Comment by michelleat** [comment]
> How did you test this action-based interface? Do you have a test node that you called this with or did you just call it on the terminal 

I tested it only in the terminal so far

File: CMakeLists.txt (hunk 1 above)
Comment (michelleat) [reply CMakeLists.txt:57]: yeah that's the original rover code.  should I change it?

Comment (pvishal) [reply CMakeLists.txt:57]: No don't bother in that case. Let's collect those cleanups for after the demo.

File: .gitignore (hunk 2 above)
Comment (michelleat) [reply .gitignore:1]: @khrisnakamarga do you know which .gitignore I should use?

**Comment by michelleat** [comment]
> How did you test this action-based interface? Do you have a test node that you called this with or did you just call it on the terminal?

@khrisnakamarga @pvishal @dean4ta In my last commit I added a test node that sends action goals for the action server, take a look!

File: package.xml (hunk 4 above)
Comment (michelleat) [reply package.xml]: done!

**Pull request merged**